src/patcher/java/net/minecraftforge/gradle/patcher/PatcherPlugin.java

```diff
@@ -300,463 +298,384 @@ public void apply(@Nonnull Project project) {
+                    binPatchesTask.configure(task -> task.getPatchSets().from(fakePatchesTask.flatMap(CreateFakeSASPatches::getOutput)));
                 }
             }
 
-            applyRangeConfig.get().setExcFiles(extension.getExcs());
-            applyRangeBaseConfig.get().setExcFiles(extension.getExcs());
-
             if (!extension.getExtraMappings().isEmpty()) {
                 extension.getExtraMappings().stream().filter(e -> e instanceof File).map(e -> (File) e).forEach(e -> {
                     userdevJar.get().from(e, c -> c.into("srgs/"));
-                    userdevConfig.get().addSRG(e);
+                    userdevConfig.get().getSRGs().from(e);
                 });
-                extension.getExtraMappings().stream().filter(e -> e instanceof String).map(e -> (String) e).forEach(e -> userdevConfig.get().addSRGLine(e));
+                extension.getExtraMappings().stream().filter(e -> e instanceof String)
+                        .map(e -> (String) e).forEach(e -> userdevConfig.get().getSRGLines().add(e));
             }
 
             //UserDev Config Default Values
-            if (userdevConfig.get().getTool() == null) {
-                userdevConfig.get().setTool("net.minecraftforge:binarypatcher:" + genJoinedBinPatches.get().getResolvedVersion() + ":fatjar");
-                userdevConfig.get().setArguments("--clean", "{clean}", "--output", "{output}", "--apply", "{patch}");
-            }
-            if (userdevConfig.get().getUniversal() == null) {
-                userdevConfig.get().setUniversal(project.getGroup().toString() + ':' + universalJar.get().getArchiveBaseName().getOrNull() + ':' + project.getVersion() + ':' + universalJar.get().getArchiveClassifier().getOrNull() + '@' + universalJar.get().getArchiveExtension().getOrNull());
-            }
-            if (userdevConfig.get().getSource() == null) {
-                userdevConfig.get().setSource(project.getGroup().toString() + ':' + sourcesJar.get().getArchiveBaseName().getOrNull() + ':' + project.getVersion() + ':' + sourcesJar.get().getArchiveClassifier().getOrNull() + '@' + sourcesJar.get().getArchiveExtension().getOrNull());
-            }
-            if (!"a/".contentEquals(genPatches.get().getOriginalPrefix())) {
-                userdevConfig.get().setPatchesOriginalPrefix(genPatches.get().getOriginalPrefix());
-            }
-            if (!"b/".contentEquals(genPatches.get().getModifiedPrefix())) {
-                userdevConfig.get().setPatchesModifiedPrefix(genPatches.get().getModifiedPrefix());
-            }
+            userdevConfig.configure(task -> {
+                task.getTool().convention(genJoinedBinPatches.map(JarExec::getResolvedVersion)
+                        .map(ver -> "net.minecraftforge:binarypatcher:" + ver + ":fatjar"));
+                task.getArguments().addAll("--clean", "{clean}", "--output", "{output}", "--apply", "{patch}");
+                task.getUniversal().convention(universalJar.flatMap(t ->
```

Comment:
It can be simplified at the cost of possibly removing property metadata. The reason why this apparent callback hell exists is to at the best of my ability preserve the attached metadata to the property, so in the case that the properties are derived from tasks or other sources, Gradle can use that metadata to figure out what must be run before the task. I welcome you to try improve it, as that was my second attempt at cleaning it up (and the formatting makes it at least a bit more bearable).